state/accountsDBApiWithHistory_test.go

```diff
@@ -227,3 +231,81 @@ func TestAccountsDBApiWithHistory_GetCodeWithBlockInfo(t *testing.T) {
 		assert.Equal(t, rootHash, recreatedRootHash)
 	})
 }
+
+func TestAccountsDBApiWithHistory_GetAccountWithBlockInfoWhenHighConcurrency(t *testing.T) {
+	numTestRuns := 16
+	numRootHashes := 128
+	numCallsPerRootHash := 128
+	recreationsCounterByRootHash := make(map[string]*atomic.Counter)
+
+	for run := 0; run < numTestRuns; run++ {
+		for i := 0; i < numRootHashes; i++ {
+			rootHashAsString := fmt.Sprintf("%d", i)
+			recreationsCounterByRootHash[rootHashAsString] = &atomic.Counter{}
+		}
+
+		dummyAccount := createDummyAccountWithBalanceBytes([]byte{42, 0, 0, 0, 0, 0, 0, 0})
+		var dummyAccountMutex sync.RWMutex
+
+		accountsAdapter := &mockState.AccountsStub{
+			RecreateTrieCalled: func(rootHash []byte) error {
+				dummyAccountMutex.Lock()
+				defer dummyAccountMutex.Unlock()
+
+				// When a trie is recreated, we "add" to it a single account,
+				// having the balance correlated with the trie rootHash (for the sake of the test, for easier assertions).
+				dummyAccount = createDummyAccountWithBalanceString(string(rootHash))
+
+				// We also count the re-creation
+				counter := recreationsCounterByRootHash[string(rootHash)]
+				counter.Increment()
+				return nil
+			},
+			GetExistingAccountCalled: func(addressContainer []byte) (vmcommon.AccountHandler, error) {
+				dummyAccountMutex.RLock()
+				defer dummyAccountMutex.RUnlock()
+				return dummyAccount, nil
+			},
+		}
+
+		accountsApiWithHistory, _ := state.NewAccountsDBApiWithHistory(accountsAdapter)
+
+		var wg sync.WaitGroup
+
+		// Simulate many concurrent goroutines calling GetAccountWithBlockInfo()
+		for i := 0; i < numRootHashes; i++ {
+			for j := 0; j < numCallsPerRootHash; j++ {
+				wg.Add(1)
+
+				go func(rootHashAsInt int) {
```

Comment:
wondering how these numbers numRootHashes = 128 and numCallsPerRootHash = 128 do not crash the race detector since it can handle a maximum 8192 go routines